_dprhtml/js/search.test.js

```diff
@@ -0,0 +1,39 @@
+import DPR_G from "./globalObject";
+import { searchTipitaka, SearchType } from "./search";
+
+describe("searchTipitaka", () => {
+  beforeEach(() => {
+    $('<div/>', { id: 'finished' }).appendTo('body');
+    $('<div/>', { id: 'sbfab' }).appendTo('body');
+    $('<div/>', { id: 'sbfb' }).appendTo('body');
+    $('<div/>', { id: 'sbfbc' }).appendTo('body');
+    $('<div/>', { id: 'searchb' }).appendTo('body');
+    $('<div/>', { id: 'search-link' }).appendTo('body');
+    $('<div/>', { id: 'search-sets' }).appendTo('body');
```

Comment:
This is the minimum set of DOM elements that need to exist when `searchTipitaka` is called to avoid throwing errors akin to `Cannot read property XXX of undefined`.